assets/javascripts/discourse/components/ai-spam.gjs

```diff
@@ -0,0 +1,194 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import DToggleSwitch from "discourse/components/d-toggle-switch";
+import DTooltip from "discourse/components/d-tooltip";
+import withEventValue from "discourse/helpers/with-event-value";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import i18n from "discourse-common/helpers/i18n";
+import ComboBox from "select-kit/components/combo-box";
+
+export default class AiSpam extends Component {
+  @service siteSettings;
+  @service toasts;
+
+  @tracked
+  stats = {
+    scanned_count: 0,
+    spam_detected: 0,
+    false_positives: 0,
+    false_negatives: 0,
+    daily_data: [],
+  };
+  @tracked isEnabled = false;
+  @tracked selectedLLM = null;
+  @tracked customInstructions = "";
+
+  constructor() {
+    super(...arguments);
+    this.initializeFromModel();
+  }
+
+  @action
+  initializeFromModel() {
+    const model = this.args.model;
+    this.isEnabled = model.is_enabled;
+    this.selectedLLM = "custom:" + model.llm_id;
+    this.customInstructions = model.custom_instructions;
+    this.stats = model.stats;
+  }
+
+  get availableLLMs() {
+    return (this.args.model?.available_llms || []).map((llm) => ({
+      id: llm.id,
+      name: llm.name,
+    }));
+  }
+
+  @action
+  async toggleEnabled() {
+    try {
+      // so UI responds immediately
+      this.isEnabled = !this.isEnabled;
+      const response = await ajax("/admin/plugins/discourse-ai/ai-spam.json", {
+        type: "PUT",
+        data: { is_enabled: this.isEnabled },
+      });
+      this.isEnabled = response.is_enabled;
+    } catch (error) {
+      popupAjaxError(error);
+    }
+  }
+
+  @action
+  async updateLLM(value) {
+    this.selectedLLM = value;
+  }
+
+  @action
+  async save() {
+    const llmId = this.selectedLLM.toString().split(":")[1];
+    try {
+      await ajax("/admin/plugins/discourse-ai/ai-spam.json", {
+        type: "PUT",
+        data: {
+          llm_model_id: llmId,
+          custom_instructions: this.customInstructions,
+        },
+      });
+      this.toasts.success({
+        data: { message: i18n("discourse_ai.spam.settings_saved") },
+        duration: 2000,
+      });
+    } catch (error) {
+      popupAjaxError(error);
+    }
```

Comment:
hmmm maybe ... it would be a wider change to discourse ai though, maybe follow up and make it consistent after chatting about it on dev. Downside though is that it is less 100% clear that an error just happened.